guards/core/guard.py

```diff
@@ -0,0 +1,41 @@
+from abc import ABC, abstractmethod
+from typing import Any
+from guards.core.events import RUN_EVENT, CLOSE_EVENT
+
+
+class Guard(ABC):
+    def __init__(self, name: str = None, meta=None):
+        self.name: str = name
+        self.meta = meta
+        self.listens_on: list[str] = [RUN_EVENT, CLOSE_EVENT]
+        self.permitted: bool = False
+        self.closed: bool = False
+        self.message: Any = None
+        self.__bound: bool = False
+
+    def bind(self):
+        if self.__bound:
+            raise Exception(
```

Comment:
I know this is not `GuardException`, but it could be more special than the base `Exception`

Reply:
Done, check it, it may not be the perfect choice but I thought I should not create an exception just for this.